source/framework/core/src/TRestMetadata.cxx

```diff
@@ -656,6 +656,54 @@ Int_t TRestMetadata::LoadConfigFromBuffer() {
     return -1;
 }
 
+///////////////////////////////////////////////
+/// \brief This method will retrieve a new TRestMetadata instance of a child element
+/// of the present TRestMetadata instance. I.e. `TRestChildClass` in the following
+/// example:
+///
+/// \code
+///    <TRestThisMetadataClass ...
+///         <TRestChildClass ...>
+/// \endcode
+///
+/// Two optional arguments may help to restrict the search to a particular metadata
+/// class name and user given name.
+///
+/// - *pattern*: If a pattern value is given, then the pattern must be contained inside
+/// the metadata class name. I.e. pattern="TRestGeant4" will require that the class
+/// belongs to the geant4 library.
+///
+/// - *name*: It can be specified a specific given name.
+///
+/// Otherwise, the first child section that satisfies that it starts by `TRest` will be
+/// returned.
+///
+/// If no child element is found `nullptr` will be returned.
+///
+TRestMetadata* TRestMetadata::GetChildMetadata(std::string pattern, std::string name) {
```

Comment:
Maybe `InitializeChildMetadata()` could be a better name?

Reply:
Perhaps `InstantiateChildMetadata`?

Reply:
Yes!